xbmc/utils/EGLUtils.cpp

```diff
@@ -610,48 +611,19 @@ bool CEGLContextUtils::TrySwapBuffers()
   return (eglSwapBuffers(m_eglDisplay, m_eglSurface) == EGL_TRUE);
 }
 
-bool CEGLContextUtils::BindTextureUploadContext()
+bool CEGLContextUtils::BindSecondaryGPUContext(unsigned int id)
 {
-  if (m_eglDisplay == EGL_NO_DISPLAY || m_eglUploadContext == EGL_NO_CONTEXT)
+  if (id > m_eglSecondaryContexts.size())
```

Comment:
This range check is incorrect:
```suggestion
  if (id >= m_eglSecondaryContexts.size())
```